### PR TITLE
Change GroupNormalization default groups to 32.

### LINKED ISSUE #2240: Set GroupNormalization default `groups` parameter to 32.
Currently, the default `groups` parameter for [`tfa.layers.GroupNormalization`](https://www.tensorflow.org/addons/api_docs/python/tfa/layers/GroupNormalization) is set to 2.

According to the original [Group Normalization paper](https://arxiv.org/abs/1803.08494) it would make more sense to set that default to 32. For reference, here is `Table 3 (top)` from the paper:

| `groups` | 64 | 32 | 16 | 8 | 4 | 2 |1 (=LN) |
| --- | --- | --- | --- | --- | --- | --- | --- |
| ResNet-50 ImageNet validation error `%` | 24.6 | ***24.1*** | 24.6 | 24.4 | 24.6 | 24.7 | 25.3 |
| Difference to `groups=32` | 0.5|  - | 0.5 | 0.3 | 0.5 | 0.6 | 1.2 |

Note that most experiments in the paper were performed using `groups=32` as well.

### PR DESCRIPTION
**Pull request by aaronmondal**
This changes the default `groups` parameter to `32`, as recommended by the authors of [Group Normalization](https://arxiv.org/abs/1803.08494).

Fixes #2240

- [x] This PR addresses an already submitted issue for TensorFlow Addons

### CODE REVIEW

**Comment by aaronmondal** [comment]
To clarify: `test_regularizations` was modified to accommodate the default of 32. A group size of 32 was already tested in [`test_initializer`](https://github.com/tensorflow/addons/blob/38cdaa71ef44b078608527bc067f23a510311e80/tensorflow_addons/layers/tests/normalizations_test.py#L205)

**Review by Harsh188 — COMMENTED** [review]
Hi! Thanks for contributing. Could you add to the [documentation](https://github.com/tensorflow/addons/blob/38cdaa71ef44b078608527bc067f23a510311e80/tensorflow_addons/layers/normalizations.py#L27) by specifying the default number of groups used?

**Comment by aaronmondal** [comment]
Edited the `Arguments` section and added the source paper as well in the same way it was done for [`tf.nn.swish`](https://github.com/tensorflow/tensorflow/blob/fcc4b966f1265f466e82617020af93670141b009/tensorflow/python/ops/nn_impl.py#L512-L546). Linking to papers was done differently for  e.g. [`tf.nn.silu`](https://github.com/tensorflow/tensorflow/blob/6468a1d17ae43c161e836bd3d1af081e8e44ef0d/tensorflow/python/ops/nn_impl.py#L512), but I think it is better to display that link explicitly so users immediately see the link target.

File: tensorflow_addons/layers/normalizations.py [hunk 1]
```diff
@@ -25,6 +25,9 @@
 @tf.keras.utils.register_keras_serializable(package="Addons")
 class GroupNormalization(tf.keras.layers.Layer):
     """Group normalization layer.
+    
+    Source: "Group Normalization" (Yuxin Wu & Kaiming He, 2018)
```
Comment (Harsh188) [inline tensorflow_addons/layers/normalizations.py:29]: Reference is [mentioned](https://github.com/tensorflow/addons/blob/174fc18ed03d9a30ddbbe2a9d6c3087a4766973d/tensorflow_addons/layers/normalizations.py#L74) at the end.

**Review by Harsh188 — CHANGES_REQUESTED** [review]

File: tensorflow_addons/layers/normalizations.py (hunk 1 above)
Comment (Harsh188) [reply tensorflow_addons/layers/normalizations.py:29]: Could you modify it such that only one exists. It might be redundant otherwise.

Comment (aaronmondal) [reply tensorflow_addons/layers/normalizations.py:29]: My bad. Didn't notice it since it was out of my viewport :D

Comment (aaronmondal) [reply tensorflow_addons/layers/normalizations.py]: Ill change it again and put it at the top. Leaving it at the bottom seems to be inconsistent with the way this was done in other papers.

File: tensorflow_addons/layers/normalizations.py [hunk 2]
```diff
@@ -66,15 +70,12 @@ class GroupNormalization(tf.keras.layers.Layer):
 
     Output shape:
         Same shape as input.
-
```
Comment (aaronmondal) [inline tensorflow_addons/layers/normalizations.py:69]: I removed this as this way of referencing did not seem to be consistent with citations in other functions. It also makes it difficult to find the reference.

**Review by Harsh188 — APPROVED** [review]

**Comment by AakashKumarNain** [comment]
LGTM! Thanks @aaronmondal and @Harsh188

**Review by AakashKumarNain — APPROVED** [review]

**Comment by aaronmondal** [comment]
Thank you @Harsh188 and @AakashKumarNain for your quick review and approval :)
